app/WebParsing/UtsgJsonParser.hs

```diff
@@ -42,17 +44,54 @@ insertAllMeetings org = do
     manager <- liftIO $ newManager tlsManagerSettings
     response <- liftIO $ httpLbs request' manager
     let respBody = responseBody response
+    return respBody
 
-    -- decode the response
+-- Get the page number, page size and total number of courses from response
+getPageInfo :: ByteString -> Maybe (Int, Int, Int)
+getPageInfo respBody = do
+    jsonValue <- decode respBody
+    case jsonValue of 
```

Comment:
Using `do` notation here should let you simplify by avoiding explicit `case` statements.

The `Nothing` case should return `Nothing` (not `(-1, -1, -1)`). So something like 

```
... = do
    json <- decode respBody
    flip parseMaybe json $ \obj -> do
        ...
```

should work